parsl/monitoring/app_monitor.py

```diff
@@ -5,59 +5,87 @@
 from parsl.monitoring.db_logger import get_db_logger
 
 
-def monitor(pid, task_id, db_logger_config, run_id):
+def monitor(pid, task_id, db_logger_config_object, run_id):
+    """Internal
+    Monitors the Parsl task's resources by pointing psutil to the task's pid and watching it and its children.
+    """
+
+    if db_logger_config_object is None:
+        logger = get_db_logger()
+    else:
+        logger = get_db_logger(logger_name=run_id + str(task_id), db_logger_config_object=db_logger_config_object)
+
+    sleep_duration = db_logger_config_object.resource_loop_sleep_duration
+    time.sleep(sleep_duration)
+    # these values are simple to log. Other information is available in special formats such as memory below.
     simple = ["cpu_num", 'cpu_percent', 'create_time', 'cwd', 'exe', 'memory_percent', 'nice', 'name', 'num_threads', 'pid', 'ppid', 'status', 'username']
+    # values that can be summed up to see total resources used by task process and its children
     summable_values = ['cpu_percent', 'memory_percent', 'num_threads']
 
-    logger = get_db_logger(enable_es_logging=False) if db_logger_config is None else get_db_logger(**db_logger_config)
-    logger.info("starting monitoring for {} on {}".format(pid, os.getpid(), extra={'task_id': task_id, 'task_run_id': run_id}))
     pm = psutil.Process(pid)
     pm.cpu_percent()
 
     def to_mb(some_bytes):
```

Comment:
my personal taste would be to log the raw byte count, and deal with presentation units elsewhere. not a super big deal though. mostly I've found it irksome in practice dealing with monitoring systems where it's not clear what some number is: does my process occupy 20Gb or 20Mb? (when either might be plausible so i can't guess from context).

Reply:
That makes sense. I don't have a preference and am happy to remove it. It was initially included here to prepare the data for direct presentation via Elasticsearch/Kibana.

Reply:
I agree with Ben